packages/server/src/fhir/routes.ts

```diff
@@ -122,22 +122,27 @@ protectedRoutes.use('/job', jobRouter);
 /**
  * Returns the internal FHIR router.
  * This function will be executed on every request.
+ * @param ctx - The authenticated request context.
  * @returns The lazy initialized internal FHIR router.
  */
-function getInternalFhirRouter(): FhirRouter {
+function getInternalFhirRouter(ctx: AuthenticatedRequestContext): FhirRouter {
   if (!internalFhirRouter) {
-    internalFhirRouter = initInternalFhirRouter();
+    internalFhirRouter = initInternalFhirRouter(ctx);
   }
   return internalFhirRouter;
 }
 
 /**
  * Returns a new FHIR router.
  * This function should only be called once on the first request.
+ * @param ctx - The authenticated request context.
  * @returns A new FHIR router with all the internal operations.
  */
-function initInternalFhirRouter(): FhirRouter {
-  const router = new FhirRouter({ introspectionEnabled: getConfig().introspectionEnabled });
+function initInternalFhirRouter(ctx: AuthenticatedRequestContext): FhirRouter {
+  const router = new FhirRouter({
+    introspectionEnabled: getConfig().introspectionEnabled,
+    graphqlMaxDepth: ctx.project.systemSetting?.find((s) => s.name === 'graphqlMaxDepth')?.valueInteger,
```

Comment:
This seems incorrect as the `FhirRouter` would be initialized with this config for whichever project makes the _first_ request.

The `FhirRequest` may need to be extended to propagate this project-specific configuration into the graphql handler.

Should this configuration be project-based or should it just be a server config in which case we could retrieve similar to `introspectionEnabled`?

Reply:
Switched to `FhirRequest` config in [1a797b5](https://github.com/medplum/medplum/pull/4644/commits/1a797b598a07c6c2124d2e067f08c78c5bc634ff)